scripts/check-packages.js

```diff
@@ -0,0 +1,142 @@
+const fs = require('fs')
+const path = require('path')
+const glob = require('glob')
+const { printLog, printError, runMain } = require('./lib/execution-utils')
+const { command } = require('./lib/command')
+
+const PACKAGE_OPTIONS = {
+  'packages/worker': {
+    builtWithTypeScript: false,
+    extraFiles: [
+      // This file is not built with TypeScript and not listed by package.json entry points, so we
+      // need to list it here.
+      'esm/entries/string.d.ts',
+    ],
+  },
+  'packages/rum-slim': {
+    extraFiles: ['bundle/datadog-rum-slim.js'],
+  },
+  'packages/rum': {
+    extraFiles: ['bundle/datadog-rum.js'],
+  },
+  'packages/logs': {
+    extraFiles: ['bundle/datadog-logs.js'],
+  },
+}
+
+runMain(() => {
+  let success = true
+  for (const packagePath of glob.globSync('packages/*')) {
+    success = checkPackage(packagePath) && success
+  }
+
+  if (success) {
+    printLog('Packages check done.')
+  } else {
+    printError('Packages check failed.')
+    process.exitCode = 1
+  }
+})
+
+function checkPackage(packagePath) {
+  printLog(`Checking ${packagePath}`)
+  const actualFiles = getActualPackageFiles(packagePath).sort()
+  const expectedFiles = getExpectedPackageFiles(packagePath).sort()
+  let success = true
+
+  for (
+    let actualFilesIndex = 0, expectedFilesIndex = 0;
+    actualFilesIndex < actualFiles.length && expectedFilesIndex < expectedFiles.length;
+
+  ) {
+    const actualFile = actualFiles[actualFilesIndex]
+    const expectedFile = expectedFiles[expectedFilesIndex]
+
+    if (actualFile === expectedFile) {
+      actualFilesIndex += 1
+      expectedFilesIndex += 1
+    } else if (actualFile < expectedFile) {
+      printError(`  Extra file ${actualFile}`)
+      success = false
+      actualFilesIndex += 1
+    } else {
+      printError(`  Missing file ${expectedFile}`)
+      success = false
+      expectedFilesIndex += 1
+    }
+  }
+
+  return success
+}
+
+function getActualPackageFiles(packagePath) {
+  // Yarn behavior is a bit different from npm regarding `.npmignore` globs. Since we are publishing
+  // packages using npm through Lerna[1], let's use npm to list files here.
+  //
+  // [1]: Quoting Lerna doc: "Lerna always uses npm to publish packages."
+  // https://lerna.js.org/docs/features/version-and-publish#from-package
+  const output = command`npm pack --dry-run --json`.withCurrentWorkingDirectory(packagePath).run()
+  return JSON.parse(output)[0].files.map((file) => file.path)
+}
+
+function getExpectedPackageFiles(packagePath) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do you think we could rely on the [files](https://docs.npm™js.com/cli/v9/configuring-npm/package-json#files) npm configuration option for the expected file list to check?

Reply:
For the record: I didn't go with the `files` configuration, which is the same as `.npmignore` but reversed. This could be done in the future if we feel the need for it, but does not help this specific PR. But following your advice and after discussions, we decided to implement rules based on `.npmignore` instead of having to list every expected files.